### PR TITLE
Add support for Vim and NeoVim

### PR DESCRIPTION
**Pull request by SimenB**
Needs sindresorhus/env-editor#3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,6 +39,11 @@ const make = (files, opts) => {
 			continue;
 		}
 
+		if (['vim', 'neovim'].indexOf(editor.id) !== -1) {
+			args.push(`"+call cursor(${parsed.line}, ${parsed.column})"`, parsed.file);
```
Comment (sindresorhus) [inline index.js]: Does it really have to be wrapped in an inner `"`? Arguments are already quoted by Node.js. Can you test if it works without?

Comment (SimenB) [inline index.js]: You are totally correct, it works without the quotes. I hit an issue though, see other comment

**Comment by SimenB** [comment]
When I did this, I just tested directly in the terminal, I didn't do it through using this module. Doing so now to test your idea about the quotes, I hit a snag. It doesn't seem to spawn vim 😆

I had to do something like this:

```diff
diff --git i/index.js w/index.js
index c5452be..bed3bab 100644
--- i/index.js
+++ w/index.js
@@ -56,9 +56,11 @@ const make = (files, opts) => {
 module.exports = (files, opts) => {
 	const result = make(files, opts);
 
+	const stdio = result.tio ? 'inherit' : 'ignore';
+
 	const cp = childProcess.spawn(result.bin, result.args, {
 		detached: true,
-		stdio: 'ignore'
+		stdio
 	});
 
 	// Fallback
@@ -70,7 +72,11 @@ module.exports = (files, opts) => {
 		}
 	});
 
-	cp.unref();
+	if (result.tio){
+		cp.on('exit', process.exit);
+	} else {
+		cp.unref();
+	}
 };
 
 module.exports.make = make;
```

And set `result.tio` to be `true` of course. What do you think about adding tio as a property in `env-editor`?

**Comment by sindresorhus** [comment]
> And set result.tio to be true of course. What do you think about adding tio as a property in env-editor?

Good idea. I don't like the name `tio` though. Maybe `isTerminalEditor`?

**Comment by SimenB** [comment]
Added: sindresorhus/env-editor#4

File: index.js [hunk 2]
```diff
@@ -39,21 +39,29 @@ const make = (files, opts) => {
...
+			args.push(`+call cursor(${parsed.line}, ${parsed.column})`, parsed.file);
+			continue;
+		}
+
 		args.push(parsed.file);
 	}
 
 	return {
 		bin: editor.bin,
+		isTerminalEditor: editor.isTerminalEditor,
```
Comment (sindresorhus) [inline index.js]: You need to update the `openEditor.make()` docs about this also being returned.

Comment (SimenB) [inline index.js]: Done, and rebased

File: test.js [hunk 3]
```diff
@@ -27,28 +28,32 @@ test('editor - generic', t => {
 		args: [
 			'unicorn.js',
 			'rainbow.js'
-		]
+		],
+		isTerminalEditor: undefined
```
Comment (SimenB) [inline test.js]: Should this be coerced to false?

Comment (sindresorhus) [inline test.js]: Yes

File: index.js (hunk 2 above)
Comment (sindresorhus) [inline index.js]: Place it last, like in the readme.

**Pull request merged**